docs/content/python/guides/rbac.md

```diff
@@ -0,0 +1,326 @@
+repository.
+
+In this guide, we'll walk through building an RBAC policy for [GitClub][].
+
+[GitClub]: https://github.com/osohq/gitclub
+
+## Declare application types as actors and resources
+
+Oso makes authorization decisions by determining if an **actor** can perform an
+**action** on a **resource**:
+
+- **Actor**: who is performing the action? `User(id=1)`
+- **Action**: what are they trying to do? `"push"`
+- **Resource**: what are they doing it to? `Repository(id=2)`
+
+The first step of building an RBAC policy is telling Oso which application
+classes are **actors** and which are **resources**. Our example app has a pair
+of **resource** types that we want to control access to, `Organization` and
+`Repository`. We declare both as resources as follows:
+
+```polar
+resource Organization {}
+
+resource Repository {}
+```
+
+Our app also has a `User` class that will be our lone **actor** type:
+
+```polar
+actor User {}
+```
+
+This piece of syntax is called a *resource block*, and it performs two
+functions: it identifies the type as an **actor** or a **resource**, and it
+provides a centralized place to declare roles and permissions for that
+particular type.
+
+## Declare roles and permissions
+
+In GitClub, users can perform actions such as `"delete"`-ing an organization or
+`"push"`-ing to a repository. Users can also be assigned roles for either type
+of resource, such as the `"owner"` role for an `Organization` or the
+`"maintainer"` role for a `Repository`.
+
+Inside the curly braces of each `resource` block, we declare the roles and
+permissions for that resource:
+
+```polar
+resource Organization {
+  roles = [ "owner" ];
+}
+
+resource Repository {
+  permissions = [ "read", "push" ];
+  roles = [ "contributor", "maintainer" ];
+}
+```
+
+<!-- TODO(gj): transition -->
+
+## Grant permissions to roles
+
+Next, we're going to write *shorthand rules* that grant permissions to roles.
+For example, if we grant the `"push"` permission to the `"maintainer"` role in
+the `Repository` resource block, then a user who's been assigned the
+`"maintainer"` role for a particular repository can `"push"` to that
+repository. Here's our `Repository` resource block with a few shorthand rules
+added:
+
+```polar
+resource Repository {
+  permissions = [ "read", "push" ];
+  roles = [ "contributor", "maintainer" ];
+
+  "push" if "maintainer";
+  "read" if "maintainer";
+  "read" if "contributor";
+}
+```
+
+Shorthand rules expand to regular [Polar rules](polar-syntax#rules) when a
+policy is loaded. The `"push" if "maintainer"` shorthand rule above expands
+to:
+
+```polar
+has_permission(actor: Actor, "push", repository: Repository) if
+  has_role(actor, "maintainer", repository);
+```
+
+{{% callout "Note" "blue" %}}
+  Instances of our application's `User` class will match the `Actor`
+  [specializer](polar-syntax#specialization) because of our `actor User {}`
+  resource block declaration.
+{{% /callout %}}
+
+<!-- TODO(gj): transition -->
```

Comment:
Your transition in the next section first sentence seems fine

Reply:
f768c586